test/test_util.py

```diff
@@ -342,3 +342,74 @@ def test_fix_types(self):
             c = cs.get_default_configuration().get_dictionary()
             c_str = {k: str(v) for k, v in c.items()}
             self.assertEqual(fix_types(c_str, cs), c)
+
+    def test_generate_grid(self):
+        '''Test grid generation'''
+        cs = ConfigurationSpace(seed=1234)
+
+        cat1 = CategoricalHyperparameter(name='cat1', choices=['T', 'F'])
+        const1 = Constant(name='const1', value=4)
+        float1 = UniformFloatHyperparameter(name='float1', lower=-1, upper=1, log=False)
+        int1 = UniformIntegerHyperparameter(name='int1', lower=10, upper=100, log=True)
+        ord1 = OrdinalHyperparameter(name='ord1', sequence=['1', '2', '3'])
+
+        cs.add_hyperparameters([float1, int1, cat1, ord1, const1])
```

Comment:
Could you please check for a few extreme cases, too? Such as only numerical, only categorical, only constant, only a single hyperparameter, no hyperparameters yet?

Reply:
For the case of no hyperparameters, in `get_cartesian_product`, `itertools.product()` returns a single empty tuple element (I expected 0 elements) here: [https://github.com/RaghuSpaceRajan/ConfigSpace/blob/19144c34b4b0cce2e9df1bb66943bbd17392c818/ConfigSpace/util.pyx#L566]() which leads to a single empty Configuration being generated in the grid. Should we check for this and make sure no configuration is returned?

Reply:
> Should we check for this and make sure no configuration is returned?

Yes, I think so.